framework/tests/viewpointtest/ReceiverTypeNonNullTest.java

```diff
@@ -0,0 +1,17 @@
+// Test case for EISOP issue #782:
+// https://github.com/eisop/checker-framework/issues/782
+public class ReceiverTypeNonNullTest {
+    public ReceiverTypeNonNullTest() {}
+
+    public ReceiverTypeNonNullTest(Object... objs) {
```

Comment:
The original test case in #782 didn't use varargs. Is this needed to reproduce the problem?
If not, it would be nicer to actually test that the modified code path is executed - that is, add test cases that adapt the super constructor relative to the type of the current constructor and ensure that types match up. And again a test that shows that expected errors are raised.

Reply:
Removed varargs now. Please see latest test cases.